File: Problem15.py
# Prompt: What is the sum of the digits of the number 2^(1000)?
#
# Execute: python Problem13.py
#
# Answer: 1366
n = 2
for i in range(1, 1000):
  n *= 2

sum = 0
for i in str(n):
  sum += int(i)

print sum

File: Problem15.java
/*
* Prompt: How many ... routes are there through a 20×20 grid [only being able to move to the right and down]?
*
* Execute: javac Problem15.java; java Problem15; rm Problem15.class
*
* Answer: 137846528820
*/
public class Problem15 {
  // https://en.wikipedia.org/wiki/Binomial_coefficient
  public static void main(String[] args) {
    int gridSize = 20;
    long paths = 1;
    for (int i = 0; i < gridSize; i++) {
      paths *= (2 * gridSize) - i;
      paths /= i + 1;
    }
    System.out.println(paths);
  }
}
